src/modelmanager.cpp

```diff
@@ -839,16 +839,18 @@ Status ModelManager::configFileReloadNeeded(bool& isNeeded) {
     return StatusCode::OK;
 }
 
-void ModelManager::watcher(std::future<void> exitSignal) {
+void ModelManager::watcher(std::future<void> exitSignal, bool watchConfigFile) {
     SPDLOG_LOGGER_INFO(modelmanager_logger, "Started model manager thread");
 
     while (exitSignal.wait_for(std::chrono::seconds(watcherIntervalSec)) == std::future_status::timeout) {
         SPDLOG_LOGGER_TRACE(modelmanager_logger, "Models configuration and filesystem check cycle begin");
         std::lock_guard<std::recursive_mutex> loadingLock(configMtx);
-        bool isNeeded;
-        configFileReloadNeeded(isNeeded);
-        if (isNeeded) {
-            loadConfig(configFilename);
+        if (watchConfigFile) {
+            bool isNeeded;
+            configFileReloadNeeded(isNeeded);
```

Comment:
Just wondering, why this method does not return the bool, but overrides a parameter?